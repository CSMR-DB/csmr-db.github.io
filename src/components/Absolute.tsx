import styled, {
  StyledComponent,
  css,
  FlattenSimpleInterpolation,
} from 'styled-components'
import { ReactNode } from 'react'

export interface IAbsoluteProps {
  children: ReactNode
  $top?: string
  $left?: string
  $bottom?: string
  $right?: string
  $width?: string
  $height?: string
}

export const Absolute: StyledComponent<
  'div',
  any,
  IAbsoluteProps,
  never
> = styled.div`
  position: absolute;

  ${({
    $width: width = 'auto',
    $height: height = 'auto',
    $top: top = '0',
    $left: left = '0',
    $bottom: bottom = '0',
    $right: right = '0',
  }: IAbsoluteProps): FlattenSimpleInterpolation => css`
    width: ${width};
    min-height: ${height};
    top: ${top};
    left: ${left};
    bottom: ${bottom};
    right: ${right};
  `}
`
