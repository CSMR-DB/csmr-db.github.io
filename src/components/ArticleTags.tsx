import React from 'react'
import { StyledLink } from './StyledLink'
import styled, { StyledComponent } from 'styled-components'

const StyledP: StyledComponent<'p', any, {}, never> = styled.p`
  text-align: left !important;
`

export const ArticleTags: ({ tags }: { tags?: string[] }) => JSX.Element = ({
  tags,
}: {
  tags?: string[]
}): JSX.Element => (
  <StyledP>
    Tags:{' '}
    {tags &&
      tags.map(
        (tag: string, i: number): JSX.Element => (
          <span key={i}>
            <StyledLink
              to={`/tag/${tag.toLocaleLowerCase().replace(/\s/gi, '_')}`}
            >
              {tag}
            </StyledLink>
            {i + 1 < tags.length ? ', ' : ''}
          </span>
        )
      )}
  </StyledP>
)
