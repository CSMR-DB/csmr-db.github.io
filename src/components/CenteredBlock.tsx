import styled, { StyledComponent } from 'styled-components'
import { WidthProperty } from 'csstype'

interface ICenteredBlockProps {
  maxWidth?: WidthProperty<1>
}

export const CenteredBlock: StyledComponent<
  'div',
  any,
  ICenteredBlockProps,
  never
> = styled.div`
  max-width: ${({
    maxWidth = '48rem',
  }: ICenteredBlockProps): WidthProperty<1> => maxWidth};
  margin: 0 auto;
`
