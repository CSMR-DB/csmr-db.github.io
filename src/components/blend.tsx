import React, { ReactNode } from 'react'
import styled, { StyledComponent } from 'styled-components'
import { MixBlendModeProperty, IsolationProperty } from 'csstype'

interface IBlend {
  children: ReactNode
  mode: MixBlendModeProperty
  hoverMode?: MixBlendModeProperty
  isolation?: boolean
}

const StyledDiv: StyledComponent<'div', any, IBlend, never> = styled.div`
  position: relative;
  isolation: ${(props: IBlend): IsolationProperty =>
    props.isolation ? 'isolate' : 'auto'};
  overflow: hidden; /** to make sure the containing element has no children overlapping parent's sibling elements */

  & > :first-child {
    position: relative !important;
  }

  :hover > :first-child {
    transform: scale(1.1);
  }

  & > * {
    mix-blend-mode: ${(props: IBlend): MixBlendModeProperty => props.mode};
    position: absolute !important; /** to work with gatsby-image, adding !important is a necessary override */
    width: 100%;
    height: 100%;
    top: 0;
    transition: all 0.25s ease-in-out;

    :hover {
      mix-blend-mode: ${(props: IBlend): MixBlendModeProperty =>
        props.hoverMode || props.mode};
    }
  }
`

export const Blend: ({
  children,
  mode,
  hoverMode,
  isolation,
}: IBlend) => JSX.Element = ({
  children,
  mode,
  hoverMode,
  isolation = false,
}: IBlend): JSX.Element => (
  <StyledDiv mode={mode} isolation={isolation} hoverMode={hoverMode}>
    {children}
  </StyledDiv>
)
