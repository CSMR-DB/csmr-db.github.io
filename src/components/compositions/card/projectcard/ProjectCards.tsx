import React from 'react'

import { IProjectsAllMarkdownRemark, IProjectsEdge } from '../../../../types/interfaces'

import { ProjectCard } from './ProjectCard'

export function ProjectCards({
  edges,
}: IProjectsAllMarkdownRemark): JSX.Element {
  return (
    <>
      {edges.map(
        (
          { node: { frontmatter, timeToRead } }: IProjectsEdge,
          i: number
        ): JSX.Element => (
          <ProjectCard
            title={frontmatter.title}
            tags={frontmatter.tags}
            video={frontmatter.featuredVideo}
            image={frontmatter.featuredImage}
            body={frontmatter.excerpt}
            timeToRead={timeToRead}
            path={frontmatter.path}
            date={frontmatter.date}
            key={i}
            index={i}
          />
        )
      )}
    </>
  )
}
