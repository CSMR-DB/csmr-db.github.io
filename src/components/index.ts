export { Absolute } from './Absolute'
export { ArticleBody } from './ArticleBody'
export { ArticleTags } from './ArticleTags'
export { Blend } from './Blend'
export {
  Card,
  CardBody,
  CardFooter,
  CardHeader,
  CardHeaderFeaturedImage,
  CardHeaderIconWrapper,
  CardHeaderText,
} from './Card'
export { CardHeaderImageOrVideo } from './CardHeaderImageOrVideo'
export { CenteredBlock } from './CenteredBlock'
export { ClippedBtn } from './ClippedBtn'
export { ContentSeparator } from './ContentSeparator'
export { CTAButton } from './CTAButton'
export { DateFormatted } from './DateFormatted'
export { DesktopNav } from './DesktopNav'
export { DynamicIcon } from './DynamicIcon'
export { DynamicImage } from './DynamicImage'
export { ExperienceCard } from './ExperienceCard'
export { Excerpt } from './Excerpt'
export { Filter } from './Filter'
export { Fixed } from './Fixed'
export { Flex } from './Flex'
export { Footer } from './Footer'
export { Grid } from './Grid'
export { Header } from './Header'
export { Heart } from './Heart'
export { Image } from './Image'
export { ImageContainerCircle } from './ImageContainerCircle'
export { ImageDot } from './ImageDot'
export { Layout } from './Layout'
export { Logo } from './Logo'
export { MenuTile } from './MenuTile'
export { ProjectCard } from './ProjectCard'
export { ReadingTime } from './ReadingTime'
export { Relative } from './Relative'
export {
  ScrollIconWrapper,
  AnimatedScroller,
  ScrollText,
} from './ScrollIndicator'
export { SEO } from './SEO'
export { SkillCard } from './SkillCard'
export { SocialButton } from './SocialButton'
export { StyledIFrame } from './StyledIFrame'
export { StyledLink, StyledA } from './StyledLink'
export { WASDNav } from './WASDNav'
