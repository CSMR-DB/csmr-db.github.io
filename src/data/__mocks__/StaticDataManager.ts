import { ExperienceData } from './../graphql/experienceDataProvider'
import { DynamicImages } from './../graphql/dynamicImagesProvider'
import { DynamicIcons } from './../graphql/dynamicIconsProvider'
import { ProjectsData } from '../graphql/projectsDataProvider'
import { SkillsetData } from '../graphql/skillsetDataProvider'
import { GameWallpapers } from '../graphql/gameWallpapersProvider'
import { SeriesPosters } from '../graphql/seriesPostersProvider'
import {
  SiteMetadata,
  SkillsetAllMarkdownRemark,
  ImageSharpAllFiles,
  ProjectsAllMarkdownRemark,
  ProjectMarkdownEdge,
  SkillsetMarkdownEdge,
  ExperienceAllMarkdownRemark,
} from '../../types/graphql.types'
import { PhotographySquares } from '../graphql/photographySquaresProvider'
import { PhotographyData } from '../graphql/photographyDataProvider'
import { Tool } from '../objects/toolsProvider'
import { AStaticDataManager } from '../StaticDataManager'
import { RouteObject } from '../objects/routesProvider'
import { Course } from '../objects/coursesProvider'
import { Contact } from '../objects/contactsProvider'
import { DefaultTheme } from 'styled-components'
import { mockSVGs } from './tools.mocks'

const mockImageCollection: ImageSharpAllFiles = {
  edges: [
    {
      node: {
        name: 'Photo',
        relativePath: '/photo.jpg',
        childImageSharp: {
          fluid: {
            aspectRatio: 1,
            sizes: '',
            src: '/photo.jpg',
            srcSet: '/photo.jpg',
          },
        },
      },
    },
  ],
}

const mockProgrammingSkillsetMdx: SkillsetAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'React',
          excerpt: 'I love JSX',
          favorite: true,
          icon: '',
          skillColor: 'red',
          time: 1337,
        },
        generatedRoute: '/skillset/react',
      },
    },
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Jest',
          excerpt: 'It Jest needs to be done',
          favorite: true,
          icon: '',
          skillColor: 'blue',
          time: 1336,
        },
        generatedRoute: '/skillset/jest',
      },
    },
  ],
}

const mockProgrammingProjectsMdx: ProjectsAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Portfolio',
          excerpt:
            'Gatsby, TypeScript, React, GraphQL, GSAP, Styled-Components, Jest, Enzyme',
          favorite: true,
          tags: [
            'Gatsby',
            'TypeScript',
            'React',
            'GraphQL',
            'GSAP',
            'Styled-Components',
            'Jest',
            'Enzyme',
          ],
        },
        generatedRoute: '/projects/portfolio',
      },
    },
    {
      node: {
        frontmatter: {
          category: 'Programming',
          date: '1994-01-08T00:00:00+0000',
          title: 'Loopy',
          excerpt: 'I made a fantastic hula-loop once',
          favorite: true,
          tags: ['Hula', 'Loop', 'Perfection'],
        },
        generatedRoute: '/projects/portfolio',
      },
    },
  ],
}

const mockGraphicDesignProjectsMdx: ProjectsAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Graphic Design',
          date: '1994-01-08T00:00:00+0000',
          title: 'Circle',
          excerpt: 'I drew a fantastic circle once',
          favorite: true,
          tags: ['Pencil', 'Shapes', 'Drawing'],
        },
        generatedRoute: '/projects/circle',
      },
    },
  ],
}

const mockGraphicDesignSkillsetMdx: SkillsetAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          category: 'Graphic Design',
          date: '1994-01-08T00:00:00+0000',
          title: 'Affinity Designer',
          excerpt: 'It has no proper geometry operations',
          favorite: false,
          icon: '',
          skillColor: 'green',
          time: 420,
        },
        generatedRoute: '/skillset/affinity_designer',
      },
    },
  ],
}

const mockExperienceMdx: ExperienceAllMarkdownRemark = {
  edges: [
    {
      node: {
        frontmatter: {
          title: 'React',
          backgroundColor: 'blue',
          type: 'Reactor',
          dateStart: '1994-01-08T00:00:00+0000',
          dateEnd: '1994-01-09T00:00:00+0000',
          excerpt: 'JSX is just fantastic',
        },
        generatedRoute: '',
      },
    },
    {
      node: {
        frontmatter: {
          title: 'Jest',
          backgroundColor: 'red',
          type: 'Jester',
          dateStart: '1994-01-08T00:00:00+0000',
          dateEnd: '1994-01-09T00:00:00+0000',
          excerpt: 'Jest a test',
        },
        generatedRoute: '',
      },
    },
  ],
}

export class StaticDataManager implements AStaticDataManager {
  static get theme(): DefaultTheme {
    return {
      palette: {
        dark: {
          normal: '#000',
          hover: '#222',
          active: '#444',
          lens: '#666',
        },
        light: {
          normal: '#fff',
          hover: '#ddd',
          active: '#bbb',
          lens: '#999',
        },
        first: { normal: '#f00', hover: '#d00', active: '#b00', lens: '#900' },
        second: {
          normal: '#0f0',
          hover: '#0d0',
          active: '#0b0',
          lens: '#090',
        },
        third: { normal: '#00f', hover: '#00d', active: '#00b', lens: '#009' },
        filter: {
          normal: 'saturation(4)',
          hover: 'saturation(4)',
          active: 'saturation(4)',
        },
      },
      breakpoints: {
        min: {
          smartphone: 'only screen and (min-width: 600px)',
          tablet: 'only screen and (min-width: 960px)',
          desktop: 'only screen and (min-width: 1200px)',
          ultrawide: 'only screen and (min-width: 1920px)',
        },
        max: {
          smartphone: 'only screen and (max-width: 600px)',
          tablet: 'only screen and (max-width: 960px)',
          desktop: 'only screen and (max-width: 1200px)',
          ultrawide: 'only screen and (max-width: 1920px)',
        },
      },
      fonts: {
        headings: 'Arial',
        copy: 'serif',
      },
    }
  }

  static get siteMetadata(): SiteMetadata {
    return {
      author: 'Jester',
      description: 'The TestRunner',
      image: '/',
      title: 'Jest',
      url: 'https://www.jest.testrunner',
    }
  }

  static get contacts(): Contact[] {
    return [
      {
        background: 'red',
        href: 'https://mock.test.nowhere',
        iconPath: '',
        text: 'Mock Option',
      },
    ]
  }

  static get courses(): Course[] {
    return [
      {
        category: 'Development',
        description: 'Experimenting with vanilla JS & jQuery',
        name: 'WebProgramming',
      },
    ]
  }

  static get routes(): RouteObject[] {
    return [
      {
        path: '/route',
        title: 'Route',
        boundKeys: ['R', 1],
      },
    ]
  }

  static get tools(): Tool[] {
    return [
      {
        name: 'React',
        svg: mockSVGs().RandomSampleSVG,
        color: 'red',
      },
    ]
  }

  static get photographyData(): PhotographyData {
    return {
      photographyData: mockImageCollection,
    }
  }

  static get photographySquares(): PhotographySquares {
    return {
      photographySquares: mockImageCollection,
    }
  }

  static get seriesPosters(): SeriesPosters {
    return {
      seriesPosters: mockImageCollection,
    }
  }

  static get gameWallpapers(): GameWallpapers {
    return {
      gameWallpapers: mockImageCollection,
    }
  }

  static get skillsetData(): SkillsetData {
    return {
      all: { ...mockProgrammingSkillsetMdx, ...mockGraphicDesignSkillsetMdx },
      top: {
        edges: [
          ...mockProgrammingSkillsetMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: SkillsetMarkdownEdge): boolean => favorite === true
          ),
          ...mockGraphicDesignSkillsetMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: SkillsetMarkdownEdge): boolean => favorite === true
          ),
        ],
      },
      allProgramming: mockProgrammingSkillsetMdx,
      allGraphicDesign: mockGraphicDesignSkillsetMdx,
    }
  }

  static get projectsData(): ProjectsData {
    return {
      all: { ...mockGraphicDesignProjectsMdx, ...mockProgrammingProjectsMdx },
      top: {
        edges: [
          ...mockGraphicDesignProjectsMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: ProjectMarkdownEdge): boolean => favorite === true
          ),
          ...mockProgrammingProjectsMdx.edges.filter(
            ({
              node: {
                frontmatter: { favorite },
              },
            }: ProjectMarkdownEdge): boolean => favorite === true
          ),
        ],
      },
      allProgramming: mockProgrammingProjectsMdx,
      allGraphicDesign: mockGraphicDesignProjectsMdx,
    }
  }

  static get experienceData(): ExperienceData {
    return {
      experienceData: mockExperienceMdx,
    }
  }

  static get dynamicIcons(): DynamicIcons {
    return {
      dynamicIcons: mockImageCollection,
    }
  }

  static get dynamicImages(): DynamicImages {
    return {
      dynamicImages: mockImageCollection,
    }
  }
}
