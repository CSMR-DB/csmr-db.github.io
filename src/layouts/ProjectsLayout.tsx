import React from 'react'
import { Reveal } from 'react-gsap'

import { Layout } from '../components/Layout'
import { Grid } from '../components/Grid'
import { ProjectsAllMarkdownRemark } from '../types/graphql.types'
import { CenteredBlock } from '../components/CenteredBlock'
import { ContentSeparator } from '../components/ContentSeparator'
import { ProjectCards } from '../components/compositions/card/projectcard/ProjectCards'

export interface IProjectsLayoutProps {
  programmingData: ProjectsAllMarkdownRemark
  graphicDesignData: ProjectsAllMarkdownRemark
}

export function ProjectsLayout({
  programmingData,
  graphicDesignData,
}: IProjectsLayoutProps): JSX.Element {
  return (
    <Layout>
      <ContentSeparator>
        <CenteredBlock>
          <h1>Hey! These are some of my projects</h1>
          <p>
            In chronological order, from newest to oldest. These are some
            projects I worked on that I am proud of, that taught me a lot and /
            or got me some decent grades.
          </p>
        </CenteredBlock>
        <Grid
          $columns={'repeat(auto-fill, minmax(32rem, 1fr))'}
          $maxWidth={'100rem'}
        >
          <Reveal>
            <ProjectCards edges={programmingData.edges} />
          </Reveal>
        </Grid>
      </ContentSeparator>
      <ContentSeparator>
        <CenteredBlock>
          <h1>I have also done some Graphic Design stuff</h1>
          <p>
            Not everything in here has to do with programming. I also included
            some projects I made using Adobe Illustrator and After Effects. The
            first projects were part of courses in a timeframe of about 10
            weeks. That meant going from 0 - 100 in roughly 10 weeks, allocating
            time for other courses as well (about 4 at a time).
          </p>
        </CenteredBlock>
        <Grid
          $columns={'repeat(auto-fill, minmax(32rem, 1fr))'}
          $maxWidth={'100rem'}
        >
          <Reveal>
            <ProjectCards edges={graphicDesignData.edges} />
          </Reveal>
        </Grid>
      </ContentSeparator>
    </Layout>
  )
}
