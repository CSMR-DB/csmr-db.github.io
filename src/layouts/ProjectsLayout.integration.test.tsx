import React, { Component } from 'react'
import { ThemeProvider } from 'styled-components'
import { create, ReactTestRenderer } from 'react-test-renderer'
import { mount, ReactWrapper } from 'enzyme'

import { StaticDataManager } from '../data/StaticDataManager'

import { ProjectsLayout } from './ProjectsLayout'
import { Layout } from '../components/Layout'
import { CenteredBlock } from '../components/CenteredBlock'
import { Grid } from '../components/Grid'
import { ProjectCards } from '../components/compositions/card/projectcard/ProjectCards'

describe('ProjectsLayout', (): void => {
  it('should match snapshot', (): void => {
    const comp: ReactTestRenderer = create(
      <ThemeProvider theme={StaticDataManager.theme}>
        <ProjectsLayout
          programmingData={StaticDataManager.projectsData.allProgramming}
          graphicDesignData={StaticDataManager.projectsData.allGraphicDesign}
        />
      </ThemeProvider>
    )

    expect(comp).toMatchSnapshot()
  })

  it('should render internally', (): void => {
    const comp: ReactWrapper<any, Readonly<{}>, Component<{}, {}, any>> = mount(
      <ThemeProvider theme={StaticDataManager.theme}>
        <ProjectsLayout
          programmingData={StaticDataManager.projectsData.allProgramming}
          graphicDesignData={StaticDataManager.projectsData.allGraphicDesign}
        />
      </ThemeProvider>
    )

    expect(comp.find(Layout).props().isLandingPage).not.toEqual(true)
    expect(comp.find(Layout).find(Grid).length).toEqual(2)
    expect(comp.find(Layout).find(CenteredBlock).length).toEqual(2)

    expect(
      comp.find(Layout).find(CenteredBlock).at(0).find('h1').text()
    ).toEqual(expect.stringContaining('my projects'))
    expect(comp.find(Layout).find(Grid).at(0).props()).toMatchObject({
      $columns: 'repeat(auto-fill, minmax(32rem, 1fr))',
      $maxWidth: '100rem',
    })
    expect(
      comp.find(Layout).find(Grid).at(0).find(ProjectCards).length
    ).toEqual(1)
    expect(
      comp.find(Layout).find(Grid).at(0).find(ProjectCards).props().edges
    ).toEqual(StaticDataManager.projectsData.allProgramming.edges)

    expect(
      comp.find(Layout).find(CenteredBlock).at(1).find('h1').text()
    ).toEqual(expect.stringContaining('Graphic Design'))
    expect(comp.find(Layout).find(Grid).at(1).props()).toMatchObject({
      $columns: 'repeat(auto-fill, minmax(32rem, 1fr))',
      $maxWidth: '100rem',
    })
    expect(
      comp.find(Layout).find(Grid).at(1).find(ProjectCards).length
    ).toEqual(1)
    expect(
      comp.find(Layout).find(Grid).at(1).find(ProjectCards).props().edges
    ).toEqual(StaticDataManager.projectsData.allGraphicDesign.edges)
  })
})
