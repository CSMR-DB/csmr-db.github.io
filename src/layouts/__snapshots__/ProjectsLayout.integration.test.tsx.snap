// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsLayout should match snapshot 1`] = `
.c1 {
  margin: 0 auto;
  padding: 1rem;
}

.c1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #f00;
}

.c1 a:hover {
  color: #d00;
}

.c0 {
  min-height: 100vh;
  background: #fff;
}

.c4 {
  display: grid;
  margin: 0 auto;
  height: auto;
  max-width: 100rem;
  gap: calc(1rem / 2);
  grid-template-columns: repeat(auto-fill,minmax(16rem,1fr));
  grid-template-rows: auto;
  -webkit-align-items: initial;
  -webkit-box-align: initial;
  -ms-flex-align: initial;
  align-items: initial;
}

.c3 {
  margin: 0 auto;
  max-width: 48rem;
}

.c2 {
  padding-bottom: 4rem;
}

.c5 {
  position: relative;
  height: 100%;
  border-radius: 0.25rem;
  display: grid;
  grid-template-areas: 'header' 'body' 'footer';
  grid-template-rows: auto 1fr auto;
  overflow: hidden;
  box-shadow: 0 0.075rem 0.075rem rgba(0,0,0,0.1), 0 0.125rem 0.125rem rgba(0,0,0,0.25);
  -webkit-transition: all 0.3s cubic-bezier(0.25,0.8,0.25,1);
  transition: all 0.3s cubic-bezier(0.25,0.8,0.25,1);
  background: #fff;
}

.c5 header img {
  -webkit-transition: all 0.2s ease-in-out !important;
  transition: all 0.2s ease-in-out !important;
}

.c5 h1 {
  font-size: 1.1rem;
  margin: 0;
}

.c5 h2 {
  font-size: 1rem;
  font-style: italic;
  margin: 0;
  color: #aaa;
  margin-bottom: 1rem;
}

.c5 p {
  text-align: left;
}

.c8 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  color: #f00 !important;
}

.c8:hover {
  color: #d00;
}

.c8.active {
  color: #b00;
}

.c11 {
  text-align: left !important;
}

.c6 {
  grid-area: 'header';
}

.c10 {
  grid-area: 'footer';
  padding: 0 2rem 2rem;
}

.c9 {
  grid-area: 'body';
  position: relative;
  padding: 2rem;
}

.c7 {
  padding: 2rem 2rem 0 2rem;
}

.c7 > h6 {
  margin-bottom: 0;
}

@media only screen and (min-width:600px) {
  .c1 {
    padding: 4rem 1rem;
  }
}

@media only screen and (min-width:600px) {
  .c4 {
    grid-template-columns: repeat(auto-fill,minmax(32rem,1fr));
    gap: 1rem;
  }
}

@media only screen and (min-width:600px) {
  .c5 {
    text-align: justify;
  }
}

@media only screen and (min-width:1200px) {
  .c5:hover {
    box-shadow: 0 0.125rem 0.25rem rgba(0,0,0,0.25), 0 0.25rem 0.25rem rgba(0,0,0,0.2);
  }

  .c5:hover header img {
    -webkit-transform: scale(1.1);
    -ms-transform: scale(1.1);
    transform: scale(1.1);
  }
}

<div
  className="c0"
>
  <main
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <h1>
          Hey! These are some of my projects
        </h1>
        <p>
          In chronological order, from newest to oldest. These are some projects I worked on that I am proud of, that taught me a lot and / or got me some decent grades.
        </p>
      </div>
      <div
        className="c4"
      >
        <article
          className="c5"
        >
          <header
            className="c6"
          >
            <div
              className="c7"
            >
              <h1>
                <a
                  className="c8"
                  href="/projects/portfolio"
                >
                  Portfolio
                </a>
              </h1>
              <h6>
                <span>
                  08 Jan 1994
                </span>
              </h6>
            </div>
          </header>
          <main
            className="c9"
          >
            <p>
              Gatsby, TypeScript, React, GraphQL, GSAP, Styled-Components, Jest, Enzyme
            </p>
          </main>
          <footer
            className="c10"
          >
            <p
              className="c11"
            >
              Tags:
               
              <span>
                <a
                  className="c8"
                  href="/skillset/gatsby"
                >
                  Gatsby
                </a>
                , 
              </span>
              <span>
                <a
                  className="c8"
                  href="/skillset/typescript"
                >
                  TypeScript
                </a>
                , 
              </span>
              <span>
                <a
                  className="c8"
                  href="/skillset/react"
                >
                  React
                </a>
                , 
              </span>
              <span>
                <a
                  className="c8"
                  href="/skillset/graphql"
                >
                  GraphQL
                </a>
                , 
              </span>
              <span>
                <a
                  className="c8"
                  href="/skillset/gsap"
                >
                  GSAP
                </a>
                , 
              </span>
              <span>
                <a
                  className="c8"
                  href="/skillset/styled-components"
                >
                  Styled-Components
                </a>
                , 
              </span>
              <span>
                <a
                  className="c8"
                  href="/skillset/jest"
                >
                  Jest
                </a>
                , 
              </span>
              <span>
                <a
                  className="c8"
                  href="/skillset/enzyme"
                >
                  Enzyme
                </a>
                
              </span>
            </p>
          </footer>
        </article>
        <article
          className="c5"
        >
          <header
            className="c6"
          >
            <div
              className="c7"
            >
              <h1>
                <a
                  className="c8"
                  href="/projects/portfolio"
                >
                  Loopy
                </a>
              </h1>
              <h6>
                <span>
                  08 Jan 1994
                </span>
              </h6>
            </div>
          </header>
          <main
            className="c9"
          >
            <p>
              I made a fantastic hula-loop once
            </p>
          </main>
          <footer
            className="c10"
          >
            <p
              className="c11"
            >
              Tags:
               
              <span>
                <a
                  className="c8"
                  href="/skillset/hula"
                >
                  Hula
                </a>
                , 
              </span>
              <span>
                <a
                  className="c8"
                  href="/skillset/loop"
                >
                  Loop
                </a>
                , 
              </span>
              <span>
                <a
                  className="c8"
                  href="/skillset/perfection"
                >
                  Perfection
                </a>
                
              </span>
            </p>
          </footer>
        </article>
      </div>
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <h1>
          I have also done some Graphic Design stuff
        </h1>
        <p>
          Not everything in here has to do with programming. I also included some projects I made using Adobe Illustrator and After Effects. The first projects were part of courses in a timeframe of about 10 weeks. That meant going from 0 - 100 in roughly 10 weeks, allocating time for other courses as well (about 4 at a time).
        </p>
      </div>
      <div
        className="c4"
      >
        <article
          className="c5"
        >
          <header
            className="c6"
          >
            <div
              className="c7"
            >
              <h1>
                <a
                  className="c8"
                  href="/projects/circle"
                >
                  Circle
                </a>
              </h1>
              <h6>
                <span>
                  08 Jan 1994
                </span>
              </h6>
            </div>
          </header>
          <main
            className="c9"
          >
            <p>
              I drew a fantastic circle once
            </p>
          </main>
          <footer
            className="c10"
          >
            <p
              className="c11"
            >
              Tags:
               
              <span>
                <a
                  className="c8"
                  href="/skillset/pencil"
                >
                  Pencil
                </a>
                , 
              </span>
              <span>
                <a
                  className="c8"
                  href="/skillset/shapes"
                >
                  Shapes
                </a>
                , 
              </span>
              <span>
                <a
                  className="c8"
                  href="/skillset/drawing"
                >
                  Drawing
                </a>
                
              </span>
            </p>
          </footer>
        </article>
      </div>
    </div>
  </main>
</div>
`;
